_posts/2017-09-05-Rust-2017-Survey-Results.md

```diff
@@ -0,0 +1,126 @@
+
+![Chart show strong support for nightly and current stable releases](/images/2017-09-05-Rust-2017-Survey-Results/which_version.png)
+
+Developers have largely opted to move to nightly or a recent stable (with some on beta), showing that developers are eager to upgrade and do so quickly.  This simplifies the support structure a bit from last year, where developers were on a wider range of versions.
+Unfortunately, despite our efforts with procedural macros and helping move crates like Serde to stable, we still have work to do to promote people moving away from the nightly Rust compiler.  This year shows an increase in nightly users, now at 1,852 votes it represents 51.6% of respondents using nightly, up from 48.8% of last year.
+
+# How we use Rust
+
+![Chart: 90.2% rustup, 18.9% linux distros, 5% homebrew, 4.7% official .msi, 3.1% official tarball, 1.4% official mac pkg](/images/2017-09-05-Rust-2017-Survey-Results/rustup.png "90.2% rustup, 18.9% linux distros, 5% homebrew, 4.7% official .msi, 3.1% official tarball, 1.4% official mac pkg")
+*(hover for more info)*
+
+One of the big success stories with Rust tooling was rustup, the Rust toolchain installer.  Last year, we saw a wide diversity in ways people were installing Rust.  This year, many of these have moved to using rustup as their main way of installing Rust, totalling now 3,205 of the responses, which moves it from last year's 52.8% to **90.2%**.
+
+![Chart: 80.9% Linux, 35.5% macOS, 31.5% Windows, 3.2% BSD-variant](/images/2017-09-05-Rust-2017-Survey-Results/platforms_on.png)
+
+Linux still features prominently as one of the main platforms Rust developers choose.  Of note, we also saw a rise in the use of Windows as a developer platform at 1,130 of the 3,588 total respondents, putting it at **31.5% of respondents**, up from 27.6% of last year.
+
+![Chart: 91.5% Linux, 46.7% Windows, 38.2% macOS, 16.8% embedded, 13.2% WebAssembly and asm.js, 9.9% Android, 8.9% BSD-variant, 5.3% Apple iOS](/images/2017-09-05-Rust-2017-Survey-Results/platforms_targetted.png)
+
+Next, we asked what platforms people were targeting with their Rust projects.  While we see a similar representation of desktop OSes here, we also see a growing range of other targets.  Android and iOS are at healthy 9.9% and 5.3% respectively, both almost **10x larger** than last year's percentages.  Embedded also has had substantial growth since last year's single-digit percentage.  As a whole, cross-compilation has grown considerably since this time last year.
+
+![Chart: 45.8% vim, 33.8% vscode, 16.1% intellij, 15.7% atom, 15.4% emacs, 12.2% sublime, 1.5% eclipse, 1.5% visual studio](/images/2017-09-05-Rust-2017-Survey-Results/editors.png)
+
+Among editors, vim remains king, though we see healthy growth in VSCode adoption at 34.1% (up from last year's 3.8%).  This growth no doubt has been helped by VSCode being one of the first platforms to get support for the [Rust Language Server](https://github.com/rust-lang-nursery/rls).
```

Comment:
It's interesting to me at least that there is pretty much a 50/50 split between IDE users and text editor users, might be worth noting. Also 'wow' at the VSCode growth :-)

Reply:
Right?